include/RAJA/policy/cuda/atomic.hpp

```diff
@@ -47,12 +47,13 @@ namespace atomic
 namespace detail
 {
 
-#if (__GNUC__ || __clang__ || __xlC__) && !__CUDA_ARCH__  // begin host compiler check
-// Silencing extraneous warnings when host compiler is activated (without device compiler) within nvcc.
-#else  // using nvcc compiler
+#if __CUDA_ARCH__ && __CUDA_ARCH__ < 350  // baseline CUDA_ARCH sm_35 check
+#warning CUDA_ARCH not set or too low, should set nvcc -arch=sm_35, COMPILING WITH DEFAULT atomicCAS
```

Comment:
Should this be #warning CUDA_ARCH set too low

Reply:
There is some odd nvcc behavior here. The following statement prints "#warning blah" when -arch=sm_xy is _not_ set:

`#if __CUDA_ARCH__ < 350`
`#warning blah`
`#endif`

Similarly, the code in question still emits a warning when -arch=sm_xy is not set (with `#if defined(__CUDA_ARCH__)`), even though it seems counterintuitive. I figured we may as well take advantage of this "feature" and help the user some more.

Reply:
I agree, I just think that the wording is confusing.
I read that as warning that CUDA_ARCH is not defined or too low. Instead of saying that CUDA_ARCH is too low because sm_xy wasn't set or was set too low on the command line.

Reply:
Should be more clear now, sorry for the unclear phrasing.

Reply:
Simplified some more and indicated the error was coming from nvcc, and not the host compiler.